packages/quantic/force-app/main/default/lwc/quanticSort/quanticSort.js

```diff
@@ -150,4 +227,44 @@ export default class QuanticSort extends LightningElement {
   get value() {
     return this.state?.sortCriteria;
   }
+
+  get hasCustomSortOptions() {
+    return this.customSortOptions.length > 0;
+  }
+
+  get hasError() {
+    return this.hasInitializationError || !!this.errorMessage;
+  }
+
+  get defaultSortOptions() {
+    return [
+      {
+        label: this.labels.relevancy,
+        value: this.headless.buildCriterionExpression(this.relevancy),
+        criterion: this.relevancy,
+      },
+      {
+        label: this.labels.newest,
+        value: this.headless.buildCriterionExpression(this.dateDescending),
+        criterion: this.dateDescending,
+      },
+      {
+        label: this.labels.oldest,
+        value: this.headless.buildCriterionExpression(this.dateAscending),
+        criterion: this.dateAscending,
+      },
+    ];
+  }
```

Comment:
IMO this should be a class property `this.defaultSortOptions` that you should initialize in the `initialize` for example.


```
export default class QuanticSort extends LightningElement {
  ...
  /* @type {SortOption[]} */
  defaultSortOptions;

  initialize = (engine) => {
    this.headless = getHeadlessBundle(this.engineId);
    ...
    this.defaultSortOptions = [
      {
        label: this.labels.relevancy,
        value: this.headless.buildCriterionExpression(this.relevancy),
        criterion: this.relevancy,
      },
      {
        label: this.labels.newest,
        value: this.headless.buildCriterionExpression(this.dateDescending),
        criterion: this.dateDescending,
      },
      {
        label: this.labels.oldest,
        value: this.headless.buildCriterionExpression(this.dateAscending),
        criterion: this.dateAscending,
      },
    ];
    ...
  }
  ...
}
```

The issue of placing it here with the `get defaultSortOptions()` is that you have no idea when `this.defaultSortOptions` could be called in the code, and this NEEDS to only be called after `this.headless` has been defined or it will crash.

Reply:
yeah that makes sense!